bin/spiced/src/lib.rs

```diff
@@ -65,73 +65,60 @@ pub async fn run(args: Args) -> Result<()> {
-                            .await
-                            .context(
-                                UnableToInitializeDataSourceSnafu {
-                                    data_source: source.clone(),
-                                },
-                            )?,
-                        )
-                    }
-                    "debug" => Box::new(datasource::debug::DebugSource {
-                        sleep_duration: Duration::from_secs(1),
-                    }),
-                    _ => UnknownDataSourceSnafu {
-                        data_source: source.clone(),
-                    }
-                    .fail()?,
-                };
-
-                let data_source = Box::leak(data_source);
-
-                df.attach(
-                    &ds.name,
-                    data_source,
-                    databackend::DataBackendType::default(),
+        let mut from_parts: Vec<&str> = ds.from.split('/').collect();
+        let source = from_parts[0];
+        from_parts.remove(0);
+        let dataset_path = from_parts.join(".");
+
+        let data_source: Box<dyn DataSource> = match source {
+            "spice.ai" => {
+                let spice_auth = auth.get(source);
+                Box::new(
+                    datasource::flight::Flight::new(
+                        spice_auth,
+                        "https://flight.spiceai.io".to_string(),
+                    )
+                    .await
+                    .context(UnableToInitializeDataSourceSnafu {
+                        data_source: source,
+                    })?,
                 )
-                .context(UnableToAttachDataSourceSnafu {
-                    data_source: source,
-                })?;
             }
-            None => {
-                df.attach_backend(&ds.name, databackend::DataBackendType::Memtable)
-                    .context(UnableToAttachDataSourceSnafu {
-                        data_source: "direct",
-                    })?;
+            "dremio" => {
+                let dremio_auth = auth.get(source);
+                Box::new(
+                    datasource::flight::Flight::new(
+                        dremio_auth,
+                        "http://dremio-4mimamg7rdeve.eastus.cloudapp.azure.com:32010".to_string(),
+                    )
+                    .await
+                    .context(UnableToInitializeDataSourceSnafu {
+                        data_source: source,
+                    })?,
+                )
             }
-        }
-
-        tracing::trace!("Loaded dataset: {}", ds.name);
+            "debug" => Box::new(datasource::debug::DebugSource {
+                sleep_duration: Duration::from_secs(1),
+            }),
+            _ => UnknownDataSourceSnafu {
+                data_source: source,
+            }
+            .fail()?,
+        };
+
+        let data_source = Box::leak(data_source);
+
+        let fq_dataset_name = format!("{}.{}", dataset_path, ds.name);
```

Comment:
Let's just attach it as the ds.name - not the fq name. As mentioned in the discussion, the fq name could support more levels than datafusion allows.

Reply:
If we do that then the dataset names will need to be unique even across sources, so I don't think that's a great solution. I think our best bet is what we do in Ducker (automatically replacing "." with "_") which we've already started doing.

Reply:
I really dislike that approach - let's find a better way. I'm fine with the constraint that dataset names need to be unique for now.

Reply:
Replacing dataset names is too much magic - especially since customers will need to know about that to make queries